### PR TITLE
save/load functionality added

### PR DESCRIPTION
**Pull request by fabiopetroni**
Added a functionality to save/load the FM model (i.e., all the parameters) on/from file.

### CODE REVIEW

**Comment by fabiopetroni** [comment]
Thank you for pointing this out.
There was a problem in linux using a string as opening file path for fstream.
I fixed the bug.

**Comment by somasdhavala** [comment]
This fork seems to write libfm model to a file and reads it. But its exact benefit and usage are not clear. 

The following questions arise:
1)
`libfm -task r -train mydata.libfm -test mydata.libfm -save_model my.model` works and it creates a model file.
But
`libfm -task r -train mydata.libfm  -save_model my.model` wont work. Basically training a model and saving it is not supported.

2) Similarly, reading a model works but still training data has to be supplied and it re-trains. then what is the purpose of loading a model. what exactly is going on
`libfm -task r -train mydata.libfm -test mydata.libfm -load_model my.model`
and it produces the following log
`Loading meta data...	
Reading FM model... 	
#Iter=  0	Train=0.82322	Test=0.82322
#Iter=  1	Train=0.810022	Test=0.804045`

But
`libfm -task r -test mydata.libfm -load_model my.model`
wont work.

In summary:
how do you use `save_model` and `load_model`?
Appreciate any help!!!

**Comment by somasdhavala** [comment]
Apparently the solution was posted on google forums. Loading and saving a model works with SGD and ALS.

For saving a model (training and testing)
`libfm -train mydata.libfm -test mydata.libfm -dim '1,1,10' -iter 100 -method 'sgd' -out pred_libfm -task r -regular '1,1,1' -learn_rate 0.1 -seed 100 -save_model fm.model`
For scoring using the above model
`libfm -train mydata.libfm -test mydata.libfm -dim '1,1,10' -iter 0 -method 'sgd' -out pred_libfm -task r -regular '1,1,1' -learn_rate 0.1 -seed 100 -load_model fm.model`
It is working
